File: backend/app/models/seeding_writer.py
"""
app/models/seeding_writer.py

种草内容仿写（seeding-writer）ORM 模型：
  - SeedingWriterConfig:   6 Prompt + 2 AI 模型配置
  - SeedingWriterProduct:  公司共享产品库
  - SeedingWriterReference: 达人维度共享素材库
"""
from sqlalchemy import BigInteger, Column, String, Text, Boolean, Integer, ForeignKey, TIMESTAMP, func

from app.core.database import Base


class SeedingWriterConfig(Base):
    """seeding-writer 工具配置（6 Prompt + 2 AI 模型，管理端可配置）"""
    __tablename__ = "seeding_writer_configs"

    id                        = Column(BigInteger, primary_key=True, autoincrement=True)
    config_key                = Column(String(64), nullable=False, unique=True)
    sp_system_prompt          = Column(Text, nullable=True)
    parse_product_prompt      = Column(Text, nullable=True)
    structure_analysis_prompt = Column(Text, nullable=True)
    ai_recommend_prompt       = Column(Text, nullable=True)
    writing_prompt            = Column(Text, nullable=True)
    iteration_prompt          = Column(Text, nullable=True)
    light_model_id            = Column(BigInteger, ForeignKey("ai_models.id", ondelete="SET NULL"), nullable=True)
    heavy_model_id            = Column(BigInteger, ForeignKey("ai_models.id", ondelete="SET NULL"), nullable=True)
    is_active                 = Column(Boolean, nullable=False, default=True)
    created_at                = Column(TIMESTAMP(timezone=True), nullable=False, server_default=func.now())
    updated_at                = Column(TIMESTAMP(timezone=True), nullable=False, server_default=func.now())


class SeedingWriterProduct(Base):
    """seeding-writer 产品库（公司共享，不按 created_by 隔离查询）"""
    __tablename__ = "seeding_writer_products"

    id                        = Column(BigInteger, primary_key=True, autoincrement=True)
    name                      = Column(Text, nullable=False)
    category                  = Column(Text, nullable=True)
    price                     = Column(Text, nullable=True)
    selling_points            = Column(Text, nullable=True)
    target_audience           = Column(Text, nullable=True)
    scenario                  = Column(Text, nullable=True)
    medical_aesthetic_anchor  = Column(Text, nullable=True)
    created_by                = Column(BigInteger, ForeignKey("users.id", ondelete="SET NULL"), nullable=True)
    created_at                = Column(TIMESTAMP(timezone=True), nullable=False, server_default=func.now())
    updated_at                = Column(TIMESTAMP(timezone=True), nullable=False, server_default=func.now())
    deleted_at                = Column(TIMESTAMP(timezone=True), nullable=True)


class SeedingWriterReference(Base):
    """seeding-writer 素材库（达人维度共享，不按 created_by 隔离查询）"""
    __tablename__ = "seeding_writer_references"

    id          = Column(BigInteger, primary_key=True, autoincrement=True)
    kol_id      = Column(BigInteger, ForeignKey("kols.id", ondelete="SET NULL"), nullable=True)
    title       = Column(Text, nullable=False)
    content     = Column(Text, nullable=False)
    type        = Column(String(32), nullable=True)
    source      = Column(String(32), nullable=True)
    likes       = Column(Integer, nullable=True)
    douyin_url  = Column(Text, nullable=True)
    created_by  = Column(BigInteger, ForeignKey("users.id", ondelete="SET NULL"), nullable=True)
    created_at  = Column(TIMESTAMP(timezone=True), nullable=False, server_default=func.now())
    updated_at  = Column(TIMESTAMP(timezone=True), nullable=False, server_default=func.now())
    deleted_at  = Column(TIMESTAMP(timezone=True), nullable=True)

File: frontend/src/types/seedingWriter.ts
// Types for 种草内容仿写（seeding-writer）

/** Step 1 达人列表项（与 persona-writer 共享同结构）*/
export interface PersonaOption {
  id: number;
  name: string;
  soul_preview: string;
  creator_name: string;
}

/** 素材库条目 */
export interface Reference {
  id: number;
  kol_id: number | null;
  title: string;
  content: string;
  type: string | null;
  source: string | null;
  likes: number | null;
  douyin_url: string | null;
  created_at: string | null;
}

/** Step 2 产品信息（表单可编辑字段）*/
export interface ProductInfo {
  name: string;
  category: string;
  price: string;
  targetAudience: string;
  sellingPoints: string;
  scenario: string;
  medicalAestheticAnchor: string;
}

/** 产品库条目（后端返回结构）*/
export interface Product {
  id: number;
  name: string;
  category: string | null;
  price: string | null;
  selling_points: string | null;
  target_audience: string | null;
  scenario: string | null;
  medical_aesthetic_anchor: string | null;
  created_by: number | null;
  created_at: string | null;
  updated_at: string | null;
}

/** 产品库分页响应 */
export interface ProductsPage {
  items: Product[];
  pagination: {
    page: number;
    page_size: number;
    total: number;
    total_pages: number;
  };
}

/** 文档解析返回（POST /products/parse-document）*/
export interface ParsedProductDocument {
  name: string;
  category: string;
  price: string;
  sellingPoints: string;
  targetAudience: string;
  scenario: string;
  medicalAestheticAnchor: string;
  _rawText: string;
}

/** chat / 卖点讨论消息 */
export interface ChatMsg {
  role: 'user' | 'assistant';
  content: string;
}

/** Step 3 fetch-video 返回 */
export interface VideoInfo {
  title: string;
  digg_count: number;
  aweme_id: string;
  play_url: string;
}

/** 管理端配置项（6 Prompt + 2 模型 + 启用）*/
export interface SeedingWriterConfig {
  id: number;
  config_key: string;
  sp_system_prompt: string | null;
  parse_product_prompt: string | null;
  structure_analysis_prompt: string | null;
  ai_recommend_prompt: string | null;
  writing_prompt: string | null;
  iteration_prompt: string | null;
  light_model_id: number | null;
  heavy_model_id: number | null;
  is_active: boolean;
  updated_at: string | null;
}

/** 历史记录项 */
export interface SeedingWriterOutput {
  id: number;
  title: string;
  content: string;
  word_count: number;
  task_id: number | null;
  created_at: string | null;
}

/** 历史记录分页响应 */
export interface OutputsPage {
  items: SeedingWriterOutput[];
  pagination: {
    page: number;
    page_size: number;
    total: number;
    total_pages: number;
  };
}

/** POST /references 请求体 */
export interface CreateReferenceRequest {
  kol_id: number;
  title: string;
  content: string;
  type?: string;
  source?: string;
  likes?: number;
}

/** POST /references/import-from-douyin 请求体 */
export interface ImportReferenceRequest {
  kol_id: number;
  share_url: string;
  type: string;
}

/** POST /products 请求体 */
export interface CreateProductRequest {
  name: string;
  category?: string;
  price?: string;
  selling_points?: string;
  target_audience?: string;
  scenario?: string;
  medical_aesthetic_anchor?: string;
}

/** PUT /products/{id} 请求体 */
export interface UpdateProductRequest extends CreateProductRequest {}

/** POST /products/extract-selling-points 请求体 */
export interface ExtractSellingPointsRequest {
  raw_text: string;
  preliminary_info: ProductInfo;
}

/** POST /analyze-structure 请求体 */
export interface AnalyzeStructureRequest {
  transcript: string;
}

/** POST /ai-recommend 请求体 */
export interface AiRecommendRequest {
  persona_id: number;
  product_id: number | null;
  reference_ids: number[];
  transcript: string;
}

/** POST /chat 请求体 */
export interface SeedingChatRequest {
  scene: 'writing' | 'iteration';
  persona_id: number;
  product_id: number | null;
  reference_ids: number[];
  transcript?: string;
  structure_analysis?: string;
  topic?: string;
  messages: ChatMsg[];
  create_job?: boolean;
}

/** POST /save-output 请求体 */
export interface SaveOutputRequest {
  content: string;
  title?: string;
  task_id?: number | null;
  topic?: string | null;
  transcript_digest?: string | null;
}

/** POST /export-word 请求体 */
export interface ExportWordRequest {
  content: string;
  filename?: string;
}
